oauthenticator/gitlab.py

```diff
@@ -124,6 +125,11 @@ class GitLabOAuthenticator(OAuthenticator):
 
         access_token = resp_json['access_token']
 
+        # memoize gitlab version for class lifetime
+        if self.gitlab_version is None:
+            self.gitlab_version = await self._get_gitlab_version(access_token)
+            self.member_api_variant = 'all/' if self.gitlab_version >= '12.4' else ''
```

Comment:
I considered this string based comparison, but its probably good and robust enough to not fail.

Reply:
Ah, I was too fixated on `11` vs `12` that I forgot about cases like `'9.5' > '12.4'`. And if the code lives for long enough, then we'd also run into `'111.0' < '12.4'` 😉 

The last problematic version, `9.5`, came out in August 2017, and `10.0` one month later. Neither version is still supported, and it's an unlikely combination to have such an old version in use for JupyterHub authentication. 

We can parse the output of `/api/v4/version` to be on the safe side

Reply:
Updated the version check to parse to a list instead